recipes/cfgfile/0.2.8.2.1/conanfile.py

```diff
@@ -0,0 +1,62 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import glob
+
+
+class CfgfileConan(ConanFile):
+    name = "cfgfile"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/igormironchik/cfgfile.git"
+    license = "MIT"
+    description = "Header-only library for reading/saving configuration files with schema defined in sources."
+    no_copy_source = True
+    generators = "cmake"
+    topics = ("conan", "cfgfile", "configuration")
+    settings = "compiler"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "3.5",
+            "apple-clang": "10"
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "14")
+
+        compiler = str(self.settings.compiler)
+        if compiler not in self._compilers_minimum_version:
+            self.output.warn("Unknown compiler, assuming it supports at least C++14")
+            return
+
+        version = tools.Version(self.settings.compiler.version)
+        if version < self._compilers_minimum_version[compiler]:
+            raise ConanInvalidConfiguration("cfgfile requires a compiler that supports at least C++14")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob(self.name + "-*/")[0]
+        os.rename(extracted_dir, self._source_subfolder)
```

Comment:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

Reply:
Nice suggestion, but 

```
tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

doesn't strip the root folder. I receive `source/source_subfolder/cfgfile-0.2.8.2.1-full/...` whereas `cfgfile-0.2.8.2.1-full` should be stripped.

Reply:
Hmm, looks like something strange in my archive...

Reply:
If create archive with leading `./` your `strip_root=True` will not strip the root. I.e. with

```
tar -czvf cfgfile-0.2.8.2.1-full.tar.gz ./cfgfile-0.2.8.2.1-full
```

Conan will fail with strip. And with

```
tar -czvf cfgfile-0.2.8.2.1-full.tar.gz cfgfile-0.2.8.2.1-full
```

All is OK. Looks like a minor issue, doesn't it?

Reply:
Why not just use the auto generated ones from GitHub? If you pass `./` then you created and extra subdirectory which explains why it failed

https://github.com/igormironchik/cfgfile/archive/refs/tags/0.2.8.2.1.tar.gz